File: java/visitor/src/main/Manager.java
package main;

import java.util.ArrayList;
import java.util.List;

public class Manager extends EmployeeBase {

  public List<EmployeeBase> Subordinates;

  public Manager() {
    Subordinates = new ArrayList<>();
  }

  @Override
  public void Accept(VisitorBase visitor) {
    visitor.Visit(this);

    for (EmployeeBase employee : Subordinates) {
      employee.Accept(visitor);
    }
  }
}

File: csharp/Visitor/Manager.cs
namespace Visitor;

public class Manager : EmployeeBase
{
    public List<EmployeeBase> Subordinates { get; set; }

    public Manager()
    {
        Subordinates = new List<EmployeeBase>();
    }

    public override void Accept(VisitorBase visitor)
    {
        visitor.Visit(this);

        foreach (var employee in Subordinates)
        {
            employee.Accept(visitor);
        }
    }
}
